test/uk/gov/hmrc/apiplatformoutboundsoap/controllers/OutboundControllerSpec.scala

```diff
@@ -81,6 +93,47 @@ class OutboundControllerSpec extends AnyWordSpec with Matchers with MockitoSugar
       messageCaptor hasCaptured MessageRequest("http://example.com/wsdl", "theOp", "<IE4N03>example</IE4N03>", addressing, Some(true))
     }
 
+    "send the message request with private headers to the outbound service" in new Setup {
+      val messageCaptor: Captor[MessageRequest] = ArgCaptor[MessageRequest]
+      when(outboundServiceMock.sendMessage(messageCaptor)).thenReturn(successful(outboundSoapMessage))
+
+      val result: Future[Result] = underTest.message()(fakeRequest.withBody(messageWithPrivateHeaders))
+      status(result) shouldBe OK
+      verify(outboundServiceMock).sendMessage(any)
+      messageCaptor hasCaptured MessageRequest("http://example.com/wsdl", "theOp", "<IE4N03>example</IE4N03>", addressing, Some(true), None, Some(privateHeaders))
+    }
+
+    "return Bad Request with too many private headers sent in the message to the outbound service" in new Setup {
```

Comment:
A request with too many headers should not reach the Outbound Service.
Perhaps better would be to assert no interactions with the outboundServiceMock and not set up any expectations on it